documentation/http_client_metrics.markdown

```diff
@@ -0,0 +1,120 @@
+---
+layout: default
+title: "Puppet Server: HTTP Client Metrics"
+canonical: "/puppetserver/latest/http_client_metrics.html"
+---
+
+HTTP client metrics available in Puppet Server 5 allows users to measure how long it takes for Puppet Server to make requests to and receive responses from other services, such as PuppetDB.
+
+## Determining metrics IDs
+
+All of these metrics are of the form `puppetlabs.<SERVER ID>.http-client.experimental.with-metric-id.<METRIC ID>.full-response`.
+
+> **Note:** The `<METRIC ID>` describes what the metric measures. A metric ID is represented in the [status endpoint](./status-api/v1/services.markdown) as an array of strings, and in the metric itself the strings are joined together with periods. For instance, the metric ID of `[puppetdb resource search]` is `puppetdb.resource.search`, so the full metric name would be `puppetlabs.<server-id>.http-client.experimental.with-metric-id.puppetdb.resource.search.full-response`.
+
+During the course of handling a Puppet agent run, Puppet Server makes several calls to other services to get or store information.
+
+-   During handling of a /puppet/v3/node request, Puppet Server issues:
+    -   a `facts find` request to PuppetDB for facts about the node, if they aren't yet cached (typically the first time it requests facts for the node). **Metric ID:** `[puppetdb facts find]`.
+-   During handling of a /puppet/v3/catalog request, Puppet Server issues several requests:
+    -   a PuppetDB `replace facts` request, to replace the facts for the agent in PuppetDB with the facts it received from the agent. **Metric ID:** `[puppetdb, command, replace_facts]`.
+    -   a PuppetDB `resource search` request, to search for resources if exported resources are used. **Metric ID:** `[puppetdb, resource, search]`.
+    -   a PuppetDB `query` request, if the `puppetdb_query` function is used in Puppet code. **Metric ID:** `[puppetdb, query]`.
+    -   a PuppetDB `replace catalog` request, to replace the catalog for the agent in PuppetDB with the newly compiled catalog. **Metric ID:** `[puppetdb, command, replace_catalog]`.
+-   During handling of a /puppet/v3/reports request, Puppet Server issues:
+    -   a PuppetDB `store report` request, to store the submitted report. **Metric ID:** `[puppetb command store_report]`.
+    -   a request to the configured `reports_url` to store the report, if the HTTP report processor is enabled. **Metric ID:** `[puppetdb report http]`.
+
+## Configuring
+
+HTTP client metrics are enabled by default, but can be disabled by setting `metrics-enabled` to `false` in the `http-client` section of [`puppetserver.conf`](./config_file_puppetserver.markdown).
+
+These metrics also depend on the `server-id` setting in the `metrics` section of `puppetserver.conf`. This defaults to `localhost` but must be changed in production.
```

Comment:
s/must be/should be?  It's best that folks change it for production to be something unique per master - in order to avoid metric naming collisions when exporting to a tool like Graphite -- but there isn't anything that will technically break if users leave it as `localhost`.  I'm probably being overly pedantic here -- happy to keep what you have if you're comfortable with it.